### ISSUE TITLE
Groups are not supported error when creating a new node deterministically

### ISSUE THREAD

**Issue by rmisio** [ipfs/js-ipfs#1946]
Version: 0.34.4
Platform: OSX
Browse: Chrome

When I try to create a new IPFS node with a specific peerId, I get a "Groups not supported" error.

![image](https://user-images.githubusercontent.com/794517/54567583-f7ef7480-4999-11e9-8473-919be5cadaf2.png)

Here's how I'm creating the node:

```
new IPFS({
  "EXPERIMENTAL": {
    "pubsub": true,
    "ipnsPubsub": true
  },
  "relay": {
    "enabled": true
  },
  "repo": "./ipfs/QmUgy68cwca6FwXJf9LnoXzyM64LwsomfWDVLQTHXRbXgw",
  "init": {
    "privateKey": "Yx/lrMILAKM2bLsS8FnfOJSYkaiBUNF3OCLuHCGCoXveyXd6x2fG393UN+YITA7YgeshWTcLHMnpUMJME4gtqw=="
  }
});
```

The `ready` handler never fires and the error comes through via the `error` handler.

**Comment by rmisio** [comment]
If I remove the `EXPERIMENTAL` and `relay` keys, then I get a different error:

![image](https://user-images.githubusercontent.com/794517/54567941-2b7ece80-499b-11e9-9b7f-393329d1be8c.png)

```
{
  "repo": "ipfs/QmWTnbv52ipaJgdjj7QiP46TJoLHpz9EChf47doYFAX9bj",
  "init": {
    "privateKey": "mLVMlgsvjFd7Q7xOA2/m9HC+aNSFkFRUVLsPY8F4sboF5Y5Tj6xQ+8G1HnfxAaf+1+p6GxCu359GIFI0oMCilg=="
  }
}
```

**Comment by rmisio** [comment]
Is the schema of the options I'm providing not correct?

**Comment by alanshaw** [comment]
@jacobheun @vasco-santos @vmx do you have any bandwidth to look into this issue? It looks like a libp2p issue here but a ipld issue here: https://discuss.ipfs.io/t/using-jsipfs-pin-add/5044

**Comment by alanshaw** [comment]
Not sure if this is relevant https://github.com/mafintosh/protocol-buffers/issues/38

**Comment by vmx** [comment]
@jacobheun @vasco-santos It would be cool if you could have a look and ping me when it comes to IPLD? If you don't have the time, let me know an I'll try to see what I can do.

**Comment by jacobheun** [comment]
The issue is that the privateKey is not supported/correct. @rmisio how did you generate the private key you're using?

This is likely due to https://github.com/libp2p/js-peer-id/issues/82.

There's some discussion at https://github.com/libp2p/js-libp2p/issues/229#issuecomment-408971074 around how to create and print out a workaround if you need to reuse your own private key

**Comment by rmisio** [comment]
@jacobheun hmm... it's the same code I'm using from a project I worked on two month back. Then it worked without issue. Has something changed regarding the private key format?

Anyhow, this is how I'm generating the key:

```
export function identityKeyFromSeed(mnemonic, bits = 4096) {
  return new Promise((resolve, reject) => {
    const bip39seed = bip39.mnemonicToSeed(mnemonic, 'Secret Passphrase');
    const hmac = sha256.hmac.create('OpenBazaar seed');
    hmac.update(bip39seed);
    const seed = new Uint8Array(hmac.array());

    keys.generateKeyPairFromSeed('ed25519', seed, bits, (err, keypair) => {
      if (!err) {
        PeerId.createFromPubKey(keys.marshalPublicKey(keypair.public),
          (err, peerId) => {
            if (err) {
              reject(err);
              return;
            }

            // todo: strip the 'Key' suffix from pub and priv
            resolve({
              peerId: peerId.toBytes(),
              publicKey: keypair.public.bytes,
              privateKey: keypair._key,
            });
          });
      } else {
        reject(err);
      }
    });
  });
}
```

```
  // need base58 peerId, base64 privateKey
  getIpfsNodeInitOpts(peerId, privateKey) {
    if (typeof peerId !== 'string') {
      throw new Error('Please provide a peerId as a string.');
    }

    if (typeof privateKey !== 'string') {
      throw new Error('Please provide a privateKey as a string.');
    }    

    return {
      EXPERIMENTAL: {
        pubsub: true,
        dht: true,
      },
      relay: {
        "enabled": true,
      },
      repo: `./ipfs/${peerId}`,
      init: { privateKey },
    }
  }
```

```
    identityKeyFromSeed(seed)
      .then(identity => {
        const peerId = createFromBytes(identity.peerId).toB58String();
        const privateKey = fromByteArray(identity.privateKey);
        const ipfsInitOpts = this.getIpfsNodeInitOpts(peerId, privateKey)
        const node = new IPFS(ipfsInitOpts);
      });
```

**Comment by rmisio** [comment]
ok, interesting, if I change the private key in the identityKeyFromSeed method to `privateKey: keypair.bytes`, it seems to work.

I think I switched to `privateKey: keypair._key`  because that worked for some other context whereas `privateKey: keypair.bytes` did not, but I really don't 100% remember why I made the switch.

Anyhow, looks like it's working for now and not using the "private" property seems safer. I'll re-comment if I discover any issue.
